src/s1reader/s1_annotation.py

```diff
@@ -0,0 +1,392 @@
+'''
+A module to load annotation files for Sentinel-1 IW SLC SAFE data
+To be used for the class "Sentinel1BurstSlc"
+'''
+
+from dataclasses import dataclass
+import datetime
+import xml.etree.ElementTree as ET
+
+import numpy as np
+from scipy.interpolate import InterpolatedUnivariateSpline
+
+@dataclass
+class AnnotationBase:
+    '''
+    A virtual base class of the inheriting annotation class i.e. Product, Calibration, and Noise.
+    Not intended for standalone use.
+    '''
+    # A parent class for annotation reader for Calibrarion, Noise, and Product
+    xml_et: ET
+    #kind:str
+
+    @classmethod
+    def _parse_scalar(cls, path_field: str, str_type: str):
+        '''Parse the scalar value in the annotation
+        path_field: field in the xml_et to parse
+        str_type: Choose how the taxts in the field will be parsed: {datetime, scalar_int, scalar_float, vector_int, vector_float, str}
+        '''
+        elem_field=cls.xml_et.find(path_field)
+        if str_type=='datetime':
+            val_out=datetime.datetime.strptime(elem_field.text,'%Y-%m-%dT%H:%M:%S.%f')
```

Comment:
```suggestion
        elem_field = cls.xml_et.find(path_field)
        if str_type == 'datetime':
            val_out = datetime.datetime.strptime(elem_field.text, '%Y-%m-%dT%H:%M:%S.%f')
```
More spaces for improved readability:
assignment operator `a=b` becomes `a = b`. There's an exception for named parameters assignment: e.g `a = b(x=7, y=9)`
comparison operators `a<b` becomes `a < b`.
between function parameters `b(x,y)` becomes `b(x, y)`.
between arithmetic operators `a+b` becomes `a + b`

Please apply everywhere in PR.